crates/uv-python/src/py_launcher.rs

```diff
@@ -1,93 +1,111 @@
-        });
+                if let Some(registry_python) = read_registry_entry(&company, &tag, &tag_key) {
+                    registry_pythons.push(registry_python);
+                }
+            }
+        }
     }
 
-    // Find the first python of the version we want in the list
-    let stdout = String::from_utf8(output.stdout).map_err(|err| Error::StatusCode {
-        message: format!("The stdout of `py --list-paths` isn't UTF-8 encoded: {err}"),
-        exit_code: output.status,
-        stdout: String::from_utf8_lossy(err.as_bytes()).trim().to_string(),
-        stderr: String::from_utf8_lossy(&output.stderr).trim().to_string(),
-    })?;
+    // The registry has no natural ordering, so we're processing the latest version first.
+    registry_pythons.sort_by(|a, b| {
+        // Highest version first (reverse), but missing versions at the bottom (regular order).
+        if let (Some(version_a), Some(version_b)) = (&a.version, &b.version) {
+            version_a.cmp(version_b).reverse().then(a.path.cmp(&b.path))
+        } else {
+            a.version
+                .as_ref()
+                .map(|version| &***version)
+                .cmp(&b.version.as_ref().map(|version| &***version))
+                .then(a.path.cmp(&b.path))
+        }
+    });
 
-    Ok(PY_LIST_PATHS
-        .captures_iter(&stdout)
-        .filter_map(|captures| {
-            let (_, [major, minor, path]) = captures.extract();
-            if let (Some(major), Some(minor)) = (major.parse::<u8>().ok(), minor.parse::<u8>().ok())
-            {
-                Some(PyListPath {
-                    major,
-                    minor,
-                    executable_path: PathBuf::from(path),
-                })
-            } else {
+    Ok(registry_pythons)
+}
+
+fn read_registry_entry(company: &str, tag: &str, tag_key: &Key) -> Option<RegistryPython> {
+    // `ExecutablePath` is mandatory for executable Pythons.
+    let Some(executable_path) = tag_key
+        .open("InstallPath")
+        .and_then(|install_path| install_path.get_value("ExecutablePath"))
+        .ok()
+        .and_then(value_to_string)
+    else {
+        debug!(
+            r"Registry Python is not executable: `Software\Python\{}\{}",
+            company, tag
+        );
+        return None;
+    };
+    debug!(
+        "Found registry Python {} {}: `{}`",
+        company, tag, executable_path
+    );
+
+    // `SysVersion` is optional.
+    let version = tag_key
+        .get_value("SysVersion")
+        .ok()
+        .and_then(|value| match value {
+            Value::String(s) => Some(s),
+            _ => None,
+        })
+        .and_then(|s| match PythonVersion::from_str(&s) {
+            Ok(version) => Some(version),
+            Err(err) => {
+                debug!(
+                    "Invalid registry Python version at {}: {}",
+                    executable_path, err
+                );
```

Comment:
Similar to above, I'd say like "Skipping Python interpreter ({}) with invalid registry version {s}: {err}"

We have a message like this elsewhere:

https://github.com/astral-sh/uv/blob/7502a963e15ae40422f8386b64f8b6c72af22d82/crates/uv-python/src/discovery.rs#L621